rosidl_generator_c/resource/idl__struct.h.em

```diff
@@ -55,25 +56,30 @@ TEMPLATE(
 from rosidl_parser.definition import Service
 }@
 @[for service in content.get_elements_of_type(Service)]@
+static const @(TYPE_HASH(idl_structure_type_to_c_typename(service.namespaced_type) + "__TYPE_VERSION_HASH", type_hash['service']))@
```

Comment:
What happens if there is more than one element of type Service in a single idl? Will this cause both of those services to have the same type_hash?

Is this even a case we need to be concerned about (I don't see it in the build space of the core, at least)?

Reply:
I am making an assumption over this whole code that there will be only one interface in the .idl that is input into this generator. Meaning `num_msg + num_srv + num_action == 1`. The `rosidl_parser` rejects any input IDL that doesn't meet this, though I'm not sure  if that's supposed to be a "forever promise"?

See https://github.com/ros2/rosidl/blob/rolling/rosidl_parser/rosidl_parser/parser.py#L256